apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -52,22 +96,86 @@ router.post(
     if (!config.openAiApiKey || !config.openAiOrganization) {
       throw new error.HttpStatusError(403, 'Not implemented (feature not available)');
     }
+
     const client = new OpenAI({
       apiKey: config.openAiApiKey,
       organization: config.openAiOrganization,
     });
 
     if (req.body.__action === 'generate_question') {
-      const jobSequenceId = await generateQuestion(
+      const result = await generateQuestion(
         client,
         res.locals.course ? res.locals.course.id : undefined,
         res.locals.authn_user.user_id,
         req.body.prompt,
       );
-      res.redirect(res.locals.urlPrefix + '/jobSequence/' + jobSequenceId);
+
+      const data = result.jobResult;
+      if (data) {
+        res.send(
+          GenerationResults(data.data.html, data.data.python, result.jobSequenceId, res.locals),
+        );
+      } else {
+        throw new error.HttpStatusError(500, `Job sequence ${result.jobSequenceId} failed.`);
+      }
```

Comment:
Nit, `data` is not a good alias for `jobResult`:

```suggestion
      const data = result.jobResult;
      if (result.jobResult) {
        res.send(
          GenerationResults(result.jobResult.data.html, result.jobResult.data.python, result.jobSequenceId, res.locals),
        );
      } else {
        throw new error.HttpStatusError(500, `Job sequence ${result.jobSequenceId} failed.`);
      }
```

You might consider having `generateQuestion` return structured data instead of the entire `jobResult` object. That is, it can return `{ jobSequenceId, htmlFileContents, pythonFileContents, ... }`.

Reply:
Changed in 6919d9f4829f12ab87c64d40f2dccf405a98d563.